mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAOTest.java

```diff
@@ -186,9 +187,10 @@ private SimpleMailboxMessage createMessage(MessageId messageId, String content,
     }
 
     private MessageWithoutAttachment toMessage(Flux<CassandraMessageDAO.MessageResult> read) throws InterruptedException, java.util.concurrent.ExecutionException {
-        return read.toStream()
-            .map(CassandraMessageDAO.MessageResult::message)
+        return read
+            .flatMap(CassandraMessageDAO.MessageResult::message)
             .map(Pair::getLeft)
+            .toStream()
```

Comment:
you probably don't need a stream at all to retrieve a value from the Flux

Reply:
Reverted